### ISSUE TITLE
IMAP Cong. 8000 00004 Software Failures

### ISSUE THREAD

**Issue by mypcisamiga** [sba1/simplemail#28]
I'm getting Guru's with an IMAP configuration.  I did a "Test" in the
Account config. and right away I get a 8000 0004 error message from
task "Simple Mail - Test Account".  This is a Requester that comes up in Workbench.
If I check my IMAP mail, I also get this.
 
This is running on an A3000 030 onboard processor (no accelerator card).  I was getting same issue in 0.44 version as well.
 
There is also a report of the OS4 version having issues as well for IMAP config on an X5000:
https://forum.amiga.org/index.php?topic=73857.msg842001#new

**Comment by polluks** [comment]
8000 0004 just means "Illegal instruction", did you try a debug version with Enforcer to locate the bug?

**Comment by mypcisamiga** [comment]
Thanks -- I will try that.

I also have MuLib tools installed but need some help in running as I have not used these.  Please let me know if you could send some info on how do it.

**Comment by mypcisamiga** [comment]
This is the debug output. I'm using AmiSSL 4.2.  It crashes on line 112 of tcpip.c

amiga-mui/tcpip.c/87 [open_socket_lib()] (SimpleMail - Test Account) => Entered function
amiga-mui/tcpip.c/101 [open_socket_lib()] (SimpleMail - Test Account) => Socket library opened 1 times
amiga-mui/tcpip.c/102 [open_socket_lib()] (SimpleMail - Test Account) => Leave (1)
imap.c/866 [imap_really_connect_and_login_to_server()] (SimpleMail - Test Account) => Entered function
amiga-mui/subthreads_amiga.c/409 [thread_handle()] (Shell Process) => Received message: 0x7df966c
tcp.c/144 [tcp_connect()] (SimpleMail - Test Account) => Entered function
tcp.c/157 [tcp_connect()] (SimpleMail - Test Account) => Got socket descriptor 0
tcp.c/173 [tcp_connect()] (SimpleMail - Test Account) => Connected to socket
amiga-mui/tcpip.c/251 [open_ssl_lib()] (SimpleMail - Test Account) => Entered function
amiga-mui/tcpip.c/87 [open_socket_lib()] (SimpleMail - Test Account) => Entered function
amiga-mui/tcpip.c/112 [o

**Comment by mypcisamiga** [comment]
Attached a MuForce output at the time of crash.

WORD READ from FFFFFFD6 (INST)                 PC: 07C45674
USP : 07E577E8 SR: 0014  (U0)(-)(-)  TCB: 07E52D00
Data: 00000000 07E5C3B7 00000000 00000000 00000000 00000000 01F949FC 00000000
Addr: 07E577EC 07E52684 07DBE474 07DBDC74 07C4B6C0 07E52684 00000000 07802284
Stck: 07DBE474 00000000 07E52684 07E5A308 07C4576A 07E52684 07E52684 07800864
Stck: 07C45806 07E52684 07DBE72C 0780E024 07BED36A 00000000 00000000 01F949FC
07c45654 :  6880                       bvc.s $7c455d6
07c45656 :  4e75                       rts
07c45658 :  48e7 0006                  movem.l a5-a6,-(a7)
07c4565c :  2a6f 000c                  movea.l $c(a7),a5
07c45660 :  226c 1c54                  movea.l $1c54(a4),a1
07c45664 :  2069 0114                  movea.l $114(a1),a0
07c45668 :  2268 0058                  movea.l $58(a0),a1
07c4566c :  42a7                       clr.l -(a7)
07c4566e :  2c69 0014                  movea.l $14(a1),a6
07c45672 :  204f                       movea.l a7,a0
07c45674 : *4eae ffd6                  jsr -$2a(a6)
07c45678 :  4fef 0004                  lea.l $4(a7),a7
07c4567c :  226c 1c54                  movea.l $1c54(a4),a1
07c45680 :  2069 0114                  movea.l $114(a1),a0
07c45684 :  2268 0058                  movea.l $58(a0),a1
07c45688 :  2c69 0018                  movea.l $18(a1),a6
07c4568c :  4eae ffd6                  jsr -$2a(a6)
07c45690 :  42ad 0014                  clr.l $14(a5)
Name: "SimpleMail - Test Account"


Exception !!   00000004     TCB: 07E52D00     CTX: 0785A1D8     SSP: 07802284
USP : 07E577E8 SR: 0014  (U0)(-)(-)  TCB: 07E52D00
Data: 00000000 07E5C3B7 00000000 00000000 00000000 00000000 01F949FC 00000000
Addr: 07E577EC 07E52684 07DBE474 07DBDC74 07C4B6C0 07E52684 00000000 07802284
Stck: 07DBE474 00000000 07E52684 07E5A308 07C4576A 07E52684 07E52684 07800864
Stck: 07C45806 07E52684 07DBE72C 0780E024 07BED36A 00000000 00000000 01F949FC
07c45654 :  6880                       bvc.s $7c455d6
07c45656 :  4e75                       rts
07c45658 :  48e7 0006                  movem.l a5-a6,-(a7)
07c4565c :  2a6f 000c                  movea.l $c(a7),a5
07c45660 :  226c 1c54                  movea.l $1c54(a4),a1
07c45664 :  2069 0114                  movea.l $114(a1),a0
07c45668 :  2268 0058                  movea.l $58(a0),a1
07c4566c :  42a7                       clr.l -(a7)
07c4566e :  2c69 0014                  movea.l $14(a1),a6
07c45672 :  204f                       movea.l a7,a0
07c45674 : *4eae ffd6                  jsr -$2a(a6)
07c45678 :  4fef 0004                  lea.l $4(a7),a7
07c4567c :  226c 1c54                  movea.l $1c54(a4),a1
07c45680 :  2069 0114                  movea.l $114(a1),a0
07c45684 :  2268 0058                  movea.l $58(a0),a1
07c45688 :  2c69 0018                  movea.l $18(a1),a6
07c4568c :  4eae ffd6                  jsr -$2a(a6)
07c45690 :  42ad 0014                  clr.l $14(a5)
Name: "SimpleMail - Test Account"

**Comment by polluks** [comment]
@mypcisamiga
"It crashes on line 112 of tcpip.c" Well, _some_ thread crashes while another prints debug on line 112 of tcpip.c.
Please update to http://aminet.net/package/util/libs/AmiSSL-4.3 first.

**Comment by mypcisamiga** [comment]
I tried amissl 4.3 and still see same issue.  MuForce hit below:


WORD READ from FFFFFFD6 (INST)                 PC: 40096C24
USP : 4032B8DC SR: 0014  (U0)(-)(-)  TCB: 40326CD8
Data: 00000000 403343F0 00000000 00000000 00000001 00000000 00000000 000003E1
Addr: 4032B8E0 4032356C 402D1694 402D15E4 07FC6360 4032356C 00000000 0780226C
Stck: 402D1694 00000000 4032356C 40330B38 40096D1A 4032356C 4032356C 0780084C
Stck: 40096DB6 4032356C 4032CDAC 07F79C64 4003E91A 00000000 00000000 00000000
Stck: 000003E1 07F79D24 4032BC54 402D15E4 07F79C64 07F81E3C 0000FFFF 00000000
Stck: 00000001 00000000 C0A80167 00000003 07800A3E 07F8217E 07F88794 07F82090
Stck: 07F96700 07F37CDC 0780084C 07F81E3C 07F85FE4 07F8217E 07F820D0 07F88794
Stck: 0000FFFF 07F88794 07F88794 07F86AC8 07F88794 07F82090 07F81E3C 07F9671E
Stck: 07F81E3C 00F80992 0780084C 07EACA12 07F37CDC 07EADC2C 00000002 0000003C
Stck: 00000000 07FAE0D4 07F96700 00000000 07F37CDC 07EABB8C 07EAB5FA 07F88794
Stck: 07F9671E 07FAE024 07F37CDC 07F37CDC 07EABC6E 07F96700 07FA0006 07EABDB4
Stck: 00000038 000000F8 0CBC0010 00000001 0000FFFF 00000038 000000F8 0CBC0018
Stck: 078009F4 0000FFFF 00000001 00000000 00001080 00001000 00001080 00000000
Stck: 07F79C64 40326CD8 07F79D24 40326CD8 07EA7C80 0780084C 07EA7B8E 0000003E
Stck: 07EA87EC 00000000 00000001 00000000 00000010 00000000 4032BC54 40331836
Stck: 40331836 4032BBB0 07EBDDC2 40331836 000007EA 7DF40000 00100000 00000000
Stck: 00000000 00010000 00000000 00100000 000007ED 9A064032 6CD807F7 9D244032
Stck: BC544032 BBF407F7 9C6407F7 9C644032 BB740000 00000000 00000000 4033193C
Stck: 40331A2C 00000000 40331814 4033193C 00008000 00000000 00000000 00000000
Stck: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 07ED9B52
Stck: 07ED9B4E 40331836 00000000 07F79CB8 00000010 07EC8588 4032BC54 07F37580
Stck: 40331814 07ECA216 07F37580 07EA8848 00001000 07EC999A 40331836 40331814
Stck: 07ECC0AA 00000000 00000000 07ED9B4E 07EA7CAA 4032BBB0 07EA7BA8 00000000
Stck: 07EA7562 07EB64EA 00000000 00000000 000003E1 07F79D24 4032BC54 07FC6360
Stck: 402D15E4 07F79C64 00000000 4032BBF4 00000010 07F79C64 07EA7CAA 00000000
Stck: 4003E5FC 4032CDAC 402D15E4 402D1694 00000000 00000000 00000000 100CB02A
Stck: 00000000 4032BD78 402D150C 4032BD80 0780E024 4032CDAC 100203E1 4A7D146D
Stck: 00000000 00000000 4001E39A 402D15E4 000003E1 4032BC54 4032BC50 4032BC60
Stck: 4032BC60 000000A0 4001C0D8 402D15E4 4001C0E0 00000008 4032C65C 4032C9F4
Stck: 07FEE458 0780E024 0780084C 078A6D20 00000000 0000004C 00000001 402D1694
Stck: 00000000 696D6170 2E676D61 696C2E63 6F6D3A20 436F6E6E 65637469 6E672E2E
Stck: 2E000000 07EA8C20 07EA7CAA 0780084C 07E9D162 00000004 100CB02A 00000008
Stck: 4009694C 4032C9F4 4009694C 078543B0 07ED7E14 40095160 00000008 4032C9F4
Stck: 4032BD38 4032C9F4 0780E024 403250B4 4009694C 00F81C0C 0780084C 40096900
PC-8: 68804E75 48E70006 2A6F000C 226C1C54 20690114 22680058 42A72C69 0014204F
PC *: 4EAEFFD6 4FEF0004 226C1C54 20690114 22680058 2C690018 4EAEFFD6 42AD0014
40096ba4 :  0098 487a fd5a             ori.l #$487afd5a,(a0)+
40096baa :  487a fd68                  pea.l $40096914(pc)
40096bae :  4eb9 4001 16cc             jsr $400116cc
40096bb4 :  487a fdea                  pea.l $400969a0(pc)
40096bb8 :  4eb9 4001 16cc             jsr $400116cc
40096bbe :  4eb9 4001 16be             jsr $400116be
40096bc4 :  4fef 0018                  lea.l $18(a7),a7
40096bc8 :  4cdf 6000                  movem.l (a7)+,a5-a6
40096bcc :  4e75                       rts
40096bce :  48e7 0116                  movem.l d7/a3/a5-a6,-(a7)
40096bd2 :  2a6f 0014                  movea.l $14(a7),a5
40096bd6 :  43fa fdd8                  lea.l $400969b0(pc),a1
40096bda :  700a                       moveq.l #$a,d0
40096bdc :  2c78 0004                  movea.l $4.w,a6
40096be0 :  4eae fdd8                  jsr -$228(a6)
40096be4 :  2640                       movea.l d0,a3
40096be6 :  7e01                       moveq.l #$1,d7
40096be8 :  200b                       move.l a3,d0
40096bea :  6714                       beq.s $40096c00
40096bec :  204d                       movea.l a5,a0
40096bee :  2c4b                       movea.l a3,a6
40096bf0 :  4eae ff2e                  jsr -$d2(a6)
40096bf4 :  2e00                       move.l d0,d7
40096bf6 :  224b                       movea.l a3,a1
40096bf8 :  2c78 0004                  movea.l $4.w,a6
40096bfc :  4eae fe62                  jsr -$19e(a6)
40096c00 :  2007                       move.l d7,d0
40096c02 :  4cdf 6880                  movem.l (a7)+,d7/a3/a5-a6
40096c06 :  4e75                       rts
40096c08 :  48e7 0006                  movem.l a5-a6,-(a7)
40096c0c :  2a6f 000c                  movea.l $c(a7),a5
40096c10 :  226c 1c54                  movea.l $1c54(a4),a1
40096c14 :  2069 0114                  movea.l $114(a1),a0
40096c18 :  2268 0058                  movea.l $58(a0),a1
40096c1c :  42a7                       clr.l -(a7)
40096c1e :  2c69 0014                  movea.l $14(a1),a6
40096c22 :  204f                       movea.l a7,a0
40096c24 : *4eae ffd6                  jsr -$2a(a6)
40096c28 :  4fef 0004                  lea.l $4(a7),a7
40096c2c :  226c 1c54                  movea.l $1c54(a4),a1
40096c30 :  2069 0114                  movea.l $114(a1),a0
40096c34 :  2268 0058                  movea.l $58(a0),a1
40096c38 :  2c69 0018                  movea.l $18(a1),a6
40096c3c :  4eae ffd6                  jsr -$2a(a6)
40096c40 :  42ad 0014                  clr.l $14(a5)
40096c44 :  2f2d 001c                  move.l $1c(a5),-(a7)
40096c48 :  2f2d 0018                  move.l $18(a5),-(a7)
40096c4c :  4eb9 4004 c216             jsr $4004c216
40096c52 :  42ad 001c                  clr.l $1c(a5)
40096c56 :  42ad 0018                  clr.l $18(a5)
40096c5a :  2ead 0024                  move.l $24(a5),(a7)
40096c5e :  2f2d 0014                  move.l $14(a5),-(a7)
40096c62 :  4eb9 4004 c216             jsr $4004c216
40096c68 :  42ad 0014                  clr.l $14(a5)
40096c6c :  42ad 0024                  clr.l $24(a5)
40096c70 :  4fef 000c                  lea.l $c(a7),a7
40096c74 :  4cdf 6000                  movem.l (a7)+,a5-a6
40096c78 :  4e75                       rts
40096c7a :  48e7 0006                  movem.l a5-a6,-(a7)
40096c7e :  2a6f 000c                  movea.l $c(a7),a5
40096c82 :  42ad 0018                  clr.l $18(a5)
40096c86 :  42ad 0014                  clr.l $14(a5)
40096c8a :  41ed 001c                  lea.l $1c(a5),a0
40096c8e :  2f08                       move.l a0,-(a7)
40096c90 :  4878 0003                  pea.l $3.w
40096c94 :  487a fd28                  pea.l $400969be(pc)
40096c98 :  4eb9 4004 c1d4             jsr $4004c1d4
40096c9e :  2b40 0018                  move.l d0,$18(a5)
40096ca2 :  4fef 000c                  lea.l $c(a7),a7
Name: "SimpleMail - Test Account"  


Exception !!   00000004     TCB: 40326CD8     CTX: 0785F030     SSP: 0780226C
USP : 4032B8DC SR: 0014  (U0)(-)(-)  TCB: 40326CD8
Data: 00000000 403343F0 00000000 00000000 00000001 00000000 00000000 000003E1
Addr: 4032B8E0 4032356C 402D1694 402D15E4 07FC6360 4032356C 00000000 0780226C
Stck: 402D1694 00000000 4032356C 40330B38 40096D1A 4032356C 4032356C 0780084C
Stck: 40096DB6 4032356C 4032CDAC 07F79C64 4003E91A 00000000 00000000 00000000
Stck: 000003E1 07F79D24 4032BC54 402D15E4 07F79C64 07F81E3C 0000FFFF 00000000
Stck: 00000001 00000000 C0A80167 00000003 07800A3E 07F8217E 07F88794 07F82090
Stck: 07F96700 07F37CDC 0780084C 07F81E3C 07F85FE4 07F8217E 07F820D0 07F88794
Stck: 0000FFFF 07F88794 07F88794 07F86AC8 07F88794 07F82090 07F81E3C 07F9671E
Stck: 07F81E3C 00F80992 0780084C 07EACA12 07F37CDC 07EADC2C 00000002 0000003C
Stck: 00000000 07FAE0D4 07F96700 00000000 07F37CDC 07EABB8C 07EAB5FA 07F88794
Stck: 07F9671E 07FAE024 07F37CDC 07F37CDC 07EABC6E 07F96700 07FA0006 07EABDB4
Stck: 00000038 000000F8 0CBC0010 00000001 0000FFFF 00000038 000000F8 0CBC0018
Stck: 078009F4 0000FFFF 00000001 00000000 00001080 00001000 00001080 00000000
Stck: 07F79C64 40326CD8 07F79D24 40326CD8 07EA7C80 0780084C 07EA7B8E 0000003E
Stck: 07EA87EC 00000000 00000001 00000000 00000010 00000000 4032BC54 40331836
Stck: 40331836 4032BBB0 07EBDDC2 40331836 000007EA 7DF40000 00100000 00000000
Stck: 00000000 00010000 00000000 00100000 000007ED 9A064032 6CD807F7 9D244032
Stck: BC544032 BBF407F7 9C6407F7 9C644032 BB740000 00000000 00000000 4033193C
Stck: 40331A2C 00000000 40331814 4033193C 00008000 00000000 00000000 00000000
Stck: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 07ED9B52
Stck: 07ED9B4E 40331836 00000000 07F79CB8 00000010 07EC8588 4032BC54 07F37580
Stck: 40331814 07ECA216 07F37580 07EA8848 00001000 07EC999A 40331836 40331814
Stck: 07ECC0AA 00000000 00000000 07ED9B4E 07EA7CAA 4032BBB0 07EA7BA8 00000000
Stck: 07EA7562 07EB64EA 00000000 00000000 000003E1 07F79D24 4032BC54 07FC6360
Stck: 402D15E4 07F79C64 00000000 4032BBF4 00000010 07F79C64 07EA7CAA 00000000
Stck: 4003E5FC 4032CDAC 402D15E4 402D1694 00000000 00000000 00000000 100CB02A
Stck: 00000000 4032BD78 402D150C 4032BD80 0780E024 4032CDAC 100203E1 4A7D146D
Stck: 00000000 00000000 4001E39A 402D15E4 000003E1 4032BC54 4032BC50 4032BC60
Stck: 4032BC60 000000A0 4001C0D8 402D15E4 4001C0E0 00000008 4032C65C 4032C9F4
Stck: 07FEE458 0780E024 0780084C 078A6D20 00000000 0000004C 00000001 402D1694
Stck: 00000000 696D6170 2E676D61 696C2E63 6F6D3A20 436F6E6E 65637469 6E672E2E
Stck: 2E000000 07EA8C20 07EA7CAA 0780084C 07E9D162 00000004 100CB02A 00000008
Stck: 4009694C 4032C9F4 4009694C 078543B0 07ED7E14 40095160 00000008 4032C9F4
Stck: 4032BD38 4032C9F4 0780E024 403250B4 4009694C 00F81C0C 0780084C 40096900
PC-8: 68804E75 48E70006 2A6F000C 226C1C54 20690114 22680058 42A72C69 0014204F
PC *: 4EAEFFD6 4FEF0004 226C1C54 20690114 22680058 2C690018 4EAEFFD6 42AD0014
40096ba4 :  0098 487a fd5a             ori.l #$487afd5a,(a0)+
40096baa :  487a fd68                  pea.l $40096914(pc)
40096bae :  4eb9 4001 16cc             jsr $400116cc
40096bb4 :  487a fdea                  pea.l $400969a0(pc)
40096bb8 :  4eb9 4001 16cc             jsr $400116cc
40096bbe :  4eb9 4001 16be             jsr $400116be
40096bc4 :  4fef 0018                  lea.l $18(a7),a7
40096bc8 :  4cdf 6000                  movem.l (a7)+,a5-a6
40096bcc :  4e75                       rts
40096bce :  48e7 0116                  movem.l d7/a3/a5-a6,-(a7)
40096bd2 :  2a6f 0014                  movea.l $14(a7),a5
40096bd6 :  43fa fdd8                  lea.l $400969b0(pc),a1
40096bda :  700a                       moveq.l #$a,d0
40096bdc :  2c78 0004                  movea.l $4.w,a6
40096be0 :  4eae fdd8                  jsr -$228(a6)
40096be4 :  2640                       movea.l d0,a3
40096be6 :  7e01                       moveq.l #$1,d7
40096be8 :  200b                       move.l a3,d0
40096bea :  6714                       beq.s $40096c00
40096bec :  204d                       movea.l a5,a0
40096bee :  2c4b                       movea.l a3,a6
40096bf0 :  4eae ff2e                  jsr -$d2(a6)
40096bf4 :  2e00                       move.l d0,d7
40096bf6 :  224b                       movea.l a3,a1
40096bf8 :  2c78 0004                  movea.l $4.w,a6
40096bfc :  4eae fe62                  jsr -$19e(a6)
40096c00 :  2007                       move.l d7,d0
40096c02 :  4cdf 6880                  movem.l (a7)+,d7/a3/a5-a6
40096c06 :  4e75                       rts
40096c08 :  48e7 0006                  movem.l a5-a6,-(a7)
40096c0c :  2a6f 000c                  movea.l $c(a7),a5
40096c10 :  226c 1c54                  movea.l $1c54(a4),a1
40096c14 :  2069 0114                  movea.l $114(a1),a0
40096c18 :  2268 0058                  movea.l $58(a0),a1
40096c1c :  42a7                       clr.l -(a7)
40096c1e :  2c69 0014                  movea.l $14(a1),a6
40096c22 :  204f                       movea.l a7,a0
40096c24 : *4eae ffd6                  jsr -$2a(a6)
40096c28 :  4fef 0004                  lea.l $4(a7),a7
40096c2c :  226c 1c54                  movea.l $1c54(a4),a1
40096c30 :  2069 0114                  movea.l $114(a1),a0
40096c34 :  2268 0058                  movea.l $58(a0),a1
40096c38 :  2c69 0018                  movea.l $18(a1),a6
40096c3c :  4eae ffd6                  jsr -$2a(a6)
40096c40 :  42ad 0014                  clr.l $14(a5)
40096c44 :  2f2d 001c                  move.l $1c(a5),-(a7)
40096c48 :  2f2d 0018                  move.l $18(a5),-(a7)
40096c4c :  4eb9 4004 c216             jsr $4004c216
40096c52 :  42ad 001c                  clr.l $1c(a5)
40096c56 :  42ad 0018                  clr.l $18(a5)
40096c5a :  2ead 0024                  move.l $24(a5),(a7)
40096c5e :  2f2d 0014                  move.l $14(a5),-(a7)
40096c62 :  4eb9 4004 c216             jsr $4004c216
40096c68 :  42ad 0014                  clr.l $14(a5)
40096c6c :  42ad 0024                  clr.l $24(a5)
40096c70 :  4fef 000c                  lea.l $c(a7),a7
40096c74 :  4cdf 6000                  movem.l (a7)+,a5-a6
40096c78 :  4e75                       rts
40096c7a :  48e7 0006                  movem.l a5-a6,-(a7)
40096c7e :  2a6f 000c                  movea.l $c(a7),a5
40096c82 :  42ad 0018                  clr.l $18(a5)
40096c86 :  42ad 0014                  clr.l $14(a5)
40096c8a :  41ed 001c                  lea.l $1c(a5),a0
40096c8e :  2f08                       move.l a0,-(a7)
40096c90 :  4878 0003                  pea.l $3.w
40096c94 :  487a fd28                  pea.l $400969be(pc)
40096c98 :  4eb9 4004 c1d4             jsr $4004c1d4
40096c9e :  2b40 0018                  move.l d0,$18(a5)
40096ca2 :  4fef 000c                  lea.l $c(a7),a7
Name: "SimpleMail - Test Account"

**Comment by Futaura** [comment]
Was this fixed? You need to also run SegTracker before MuForce, I think, otherwise the output is a little useless. It is clear from the hits that something somewhere is calling a library function, when that library has not been successfully opened (library base in A6 is NULL)

**Comment by Futaura** [comment]
-$2a(A6) is CleanupAmiSSL() if A6 is AmiSSLBase or CloseAmiSSL() if A6 is AmiSSLMasterBase. The code is hard to follow, but obviously it is broken.

**Comment by mypcisamiga** [comment]
Yes, I'm still seeing issue.  I tired AmiSSL 4.4 and still see issue. Let me get SegTracker enabled.

**Comment by polluks** [comment]
@mypcisamiga If an OpenLibrary() fails just use SnoopDos.

**Comment by mypcisamiga** [comment]
Thanks.  SnoopDOS found it.   amissl_v0097m.library or amissl_v0997g.library was attempted to be opened but I didn't have those versions installed and then after attempting to open, SimpleMail crashed.  I now have installed the "g" version and no more guru.

**Comment by craziazkowboi** [comment]
Thank you @mypcisamiga!  Your fix worked for me, and I would have been lost without your post!
